### ISSUE TITLE
RX_TIMEOUT_ERROR while stopping DataStream

### ISSUE THREAD

**Issue by poett1** [basecamelectronics/sbgc32-serial-api#2]
Hello,

sbgc32-serial-api always returns RX_TIMEOUT_ERROR after stopping a requested DataStream using SBGC32_RequestDataStream(), regardless of the rxTimeout value (I increased it to `100*SBGC_RX_WAITING` for testing). This can be reproduced using the following Patch to `examples/Linux/DemoLaunch/main.c`
[main.patch](https://github.com/basecamelectronics/sbgc32-serial-api/files/11821854/main.patch)
Apply the patch by running `patch -u main.c < main.patch`. 

Output of the code is:
```
Board Info: Receive OK!
 
******************************
The system is ready to go!
Board Version: 3.6 
Firmware Version: 2.70b6 
******************************

CMD_CONFIRM #244
CMD_CONFIRM #90
Starting Data Stream
CMD_CONFIRM #85
Stopping Data Stream
CONFIRM_TIMEOUT_ERROR!
Data Stream Interval Stop error 
```

The SBGC starts and stops the stream as expected, the API simply does not get the confirmation.
Command and answer are transmitted:
![image](https://github.com/basecamelectronics/sbgc32-serial-api/assets/65654059/ca3e92cf-d796-453e-9661-7c1a28750bae)
Answer of the SBGC:
![image](https://github.com/basecamelectronics/sbgc32-serial-api/assets/65654059/338a1f45-cea5-48e8-bbec-b48280fbb815)

Any hints on this?

Also, I'm wondering why `RealTimeDataCustom_t` does not support all the data from `RealTimeData_t ` (Realtime Data 4). This would be really helpful to limit the amount of data that is transmitted to only the information that is needed by the application.

Thanks a lot!

**Comment by qsivey** [comment]
> Hello,
> 
> sbgc32-serial-api always returns RX_TIMEOUT_ERROR after stopping a requested DataStream using SBGC32_RequestDataStream(), regardless of the rxTimeout value (I increased it to `100*SBGC_RX_WAITING` for testing). This can be reproduced using the following Patch to `examples/Linux/DemoLaunch/main.c` [main.patch](https://github.com/basecamelectronics/sbgc32-serial-api/files/11821854/main.patch) Apply the patch by running `patch -u main.c < main.patch`.
> 
> Output of the code is:
> 
> ```
> Board Info: Receive OK!
>  
> ******************************
> The system is ready to go!
> Board Version: 3.6 
> Firmware Version: 2.70b6 
> ******************************
> 
> CMD_CONFIRM #244
> CMD_CONFIRM #90
> Starting Data Stream
> CMD_CONFIRM #85
> Stopping Data Stream
> CONFIRM_TIMEOUT_ERROR!
> Data Stream Interval Stop error 
> ```
> 
> The SBGC starts and stops the stream as expected, the API simply does not get the confirmation. Command and answer are transmitted: ![image](https://user-images.githubusercontent.com/65654059/247647753-ca3e92cf-d796-453e-9661-7c1a28750bae.png) Answer of the SBGC: ![image](https://user-images.githubusercontent.com/65654059/247647843-338a1f45-cea5-48e8-bbec-b48280fbb815.png)
> 
> Any hints on this?
> 
> Also, I'm wondering why `RealTimeDataCustom_t` does not support all the data from `RealTimeData_t ` (Realtime Data 4). This would be really helpful to limit the amount of data that is transmitted to only the information that is needed by the application.
> 
> Thanks a lot!

Hello!

How do you realise Stopping Data Stream function? Now it is missed in the library.

Very soon we are planning to update the library to 1.0 version, where will that function be.
Thanks for the feedback!

**Comment by poett1** [comment]
By sending the same cmdID with intervalMs=0, as described in the API documentation:

![grafik](https://github.com/basecamelectronics/sbgc32-serial-api/assets/65654059/248a8d61-acf5-41e5-9e40-885f59c1750c)

**Comment by qsivey** [comment]
> By sending the same cmdID with intervalMs=0, as described in the API documentation:
> 
> ![grafik](https://user-images.githubusercontent.com/65654059/248748685-248a8d61-acf5-41e5-9e40-885f59c1750c.png)

Hello! Sorry for delay

Now you can try a 1.0 version. Currently it is contains the SBGC32_StopDataStream() function.
Regarding your quetion about RealTimeDataCustom_t structure. If you don't find the data you need, you can look at other realtime functions.

**Comment by poett1** [comment]
This is still an error with the latest API version and Firmware 2.71

**Comment by qsivey** [comment]
Hello!
Was your problem fixed already?
If not, please contact our support and attach the executable and config.h files.
support@basecamelectronics.com

**Comment by poett1** [comment]
[0001-added-SBGC32_StopDataStream-and-error-handling.patch](https://github.com/user-attachments/files/16845003/0001-added-SBGC32_StopDataStream-and-error-handling.patch)
No, it is not solved. You can reproduce it by applying the supplied patch to this repo and running examples/Linux/DemoLaunch.
As we already have this issue open, i would not like to open it on another channel.

**Comment by qsivey** [comment]
Hello!
I tested this problem and didn't get any error.
I used Linux Ubuntu 22 and STM32F7 and it worked well.
I can advise you to delete the "sbgcDelay(1000)" string in the streaming cycle. If you want to obtain the stream data every second with enabled data synchronization, set the intervalMs value about 120-125;
Also, keep in mind that an unfinished data stream will continue transmitting, and restarting the program will register an additional data stream, which can potentially make Rx buffer overflowed.
